flow/project/cli/config/manipulators/processor.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package manipulators
+
+import (
+	"os"
+	"regexp"
+	"strings"
+)
+
+var (
+	envRegex      *regexp.Regexp = regexp.MustCompile(`\$\{env\:(.+)\}`)
+	fileRegex     *regexp.Regexp = regexp.MustCompile(`"([^"]*)"\s*:\s*{\s*"fromFile"\s*:\s*"([^"]*)"\s*},?`)
+	trailingComma *regexp.Regexp = regexp.MustCompile(`\,\s*}`)
+)
+
+// Preprocessor is used to pre-process configuration file
+type Preprocessor struct {
+	composer *Composer
+}
+
+// NewPreprocessor creates new instance of preprocessor
+func NewPreprocessor(composer *Composer) *Preprocessor {
+	return &Preprocessor{
+		composer: composer,
+	}
+}
+
+// Run all pre-processors
+func (p *Preprocessor) Run(raw []byte) []byte {
+	rawString := string(raw)
+	rawString = p.processEnv(rawString)
```

Comment:
We said we were going to change back to the `${VAR}` syntax, right?

I think that makes more sense than the `${env:VAR}` format.

We can use the package @bjartek found for this. You can call `envsubst.String(rawString)`: https://github.com/a8m/envsubst#using-envsubst-programmatically-

Reply:
This does introduce a parser ambiguity between `${VAR}` and `${file:PATH}`.

Do you think we could remove the file loading functionality, at least for now? I think that use case is well covered by the composable config files.

Reply:
Yes ofc. I now noticed my latest changes for that weren't pushed to this branch but to the `compose-config-file` branch which was nonexisting since it was merged into this one.